guiEditor/src/components/propertyTab/propertyGrids/gui/commonControlPropertyGridComponent.tsx

```diff
@@ -70,15 +70,14 @@ export class CommonControlPropertyGridComponent extends React.Component<ICommonC
         if (value.charAt(value.length - 1) === '%') {
             percentage = true;
         }
-        else if(value.charAt(value.length - 1) === 'x' && value.charAt(value.length - 2) === 'p') {
+        else if (value.charAt(value.length - 1) === 'x' && value.charAt(value.length - 2) === 'p') {
             percentage = false;
         }
-        let newValue = value.split('').filter(function (item) {
-            return (!isNaN(parseInt(item)));
-        }).join('');
 
+        let newValue = value.replace(/[\p\x\%\-]/g, "");

```

Comment:
though valid, there is no reason breaking any of those characters . They are not special characters.
So this regex could be:

`[px%-]` instead.

Reply:
It also seems like you are trying to extract the number from the string. you can do something like this:

```javascript
"-123.543px".match(/([\d\.\,]+)/)[0] === '123.543';
```

This way you always make sure no other character is parsed.

Another question here - are you taking into account different number representations? for example a german (or french)  browser will represent the number as `123,543` (with a ',' instead of '.'). Is the browser parsing the number, or are you?

Reply:
I'm doing an parse to get rip of P, X , % and - so I can add it back in myself.

Reply:
it is safer in that case to get what you need instead of removing what you don't.